### PR TITLE
Add functionality to configure metering schedule report types

### LINKED ISSUE #4871: Extend metering schedule configuration to allow setting report type
**User story**
As a KKP administrator, I would like to get/set/update report type of the schedule configuration in the Dashboard. Possible report types are `cluster`, `namespace` and `cluster and namespace`.

**Acceptance criteria**
- Enable getting/creating/editing report type in schedule configuration

### PR DESCRIPTION
**Pull request by Waseem826**
**What this PR does / why we need it**:
Add functionality to configure report types in metering schedule configuration:

- **Schedules List:**

screenshot-localhost_8000-2023 04 10-15_35_40

- **Create Schedule Dialog**

screenshot-localhost_8000-2023 04 10-15_31_19

- **Edit Schedule Dialog:**

screenshot-localhost_8000-2023 04 10-15_31_43

**Which issue(s) this PR fixes**:

Fixes #4871 

**What type of PR is this?**

/kind feature

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change? Then add your Release Note here**:

```release-note
Configure report types in schedule configuration.
```

**Documentation**:

```documentation
TBD
```

### CODE REVIEW

**Comment by Waseem826** [comment]
@pkprzekwas do we need to cherrypick this to 2.22?

File: modules/web/src/app/dynamic/enterprise/metering/schedule-config/edit-dialog/template.html [hunk 1]
```diff
@@ -58,6 +58,26 @@
...
         </mat-hint>
       </mat-form-field>
+
+      <mat-form-field>
+        <mat-label>Report Types</mat-label>
+        <mat-select [formControlName]="controls.Types"
+                    multiple
+                    panelClass="km-multiple-values-dropdown"
+                    disableOptionCentering
+                    required>
```
Comment (ahmadhamzh) [inline modules/web/src/app/dynamic/enterprise/metering/schedule-config/edit-dialog/template.html:68]: i think you need to fetch the latest main branch in order to add `kmValueChangedIndicator` directive?

File: modules/web/src/app/dynamic/enterprise/metering/schedule-config/add-dialog/component.ts [hunk 2]
```diff
@@ -182,6 +185,7 @@ export class MeteringScheduleAddDialog implements OnInit, OnDestroy {
       schedule: this._schedule,
       interval: this._interval,
       retention: this._retention,
+      types: this.form.get(Controls.Types).value,
```
Comment (ahmadhamzh) [inline modules/web/src/app/dynamic/enterprise/metering/schedule-config/add-dialog/component.ts:188]: so all types of reports will be sending even if it's not custom Configuration ?

**Review by ahmadhamzh — COMMENTED** [review]
just couple of questions PTAL

File: modules/web/src/app/dynamic/enterprise/metering/schedule-config/add-dialog/component.ts (hunk 2 above)
Comment (Waseem826) [reply modules/web/src/app/dynamic/enterprise/metering/schedule-config/add-dialog/component.ts:188]: By default all types are selected when a schedule is created but I think here it doesn't make sense to send `types`  in payload when custom configuration option is not selected. I'll update the payload.

**Pull request merged**